### PR TITLE
`Assessment`: Fix accepting complaints changes test case status

### LINKED ISSUE #4334: Accepting a complaint turns postive test cases with 0 weight into negative ones
### Describe the bug

When accepting a complaint for a programming exercise, successful test cases which don't award points (weight 0) get changed into negative ones. As a consequence, the student sees reed empty feedback elements.

### To Reproduce

1. You need a programming exercise with test cases that don't award points (weight 0)
2. A student needs to submit a complaint.
3. A tutor accepts the complaint
4. The test cases get negative and are shown red.

### Expected behavior

positive test cases stay positive.

### Screenshots

result before the complaint (tutor view):
grafik

result after the accepted complaint (tutor view):
grafik

result after the accepted complaint (student view):
grafik

### What browsers are you seeing the problem on?

Firefox

### Additional context

Looking at the data from the API, I found out that the `positive`-attribute gets changed when accepting the complaint, from `true` to `false`. 

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [ ] I added multiple integration tests (Spring) related to the features (with a high test coverage).

### Motivation and Context
Closes #4334 (see Issue for more details)

### Description
The positive value was set based on points, with 0-point-feedback being non-positive. For test cases this is the wrong behavior, a test case is positive if its execution was successful, no matter the points.

When copying the feedback into the new result for the accepted complaint, the previous positive value now stays. Only if there is no positive-value set it will be calculated based on the points of the feedback.

### Steps for Testing
Follow the steps described in the issue.

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise with Complaints enabled

1. Setup the test case grading of the exercise to have 0-weight test cases (test cases that award 0 points when successful) 
2. Participate in the exercise (make sure that some of these 0-point tests are successful)
3. Submit a complaint as a student
4. Accept the complaint
5. Check the feedback view. The successful 0-point test cases should still be positive while failed test cases stayed negative.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots:
Old wrong behavior (failed 0 point test):
grafik

new (values stays):

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java [hunk 1]
```diff
@@ -316,7 +316,9 @@ public List<Feedback> copyFeedbackToNewResult(Result newResult, Result oldResult
     private void copyFeedbackToResult(Result result, List<Feedback> feedbacks) {
         feedbacks.forEach(feedback -> {
             Feedback newFeedback = feedback.copyFeedback();
-            newFeedback.setPositiveViaCredits();
+            if (newFeedback.isPositive() == null) {
+                newFeedback.setPositiveViaCredits();
+            }
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java:321]: Do we actually need this in this place or can we integrate it inside of copyFeedback?

Comment (simon-hng) [reply src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java:321]: I agree, a boolean being `null` is IMO something that should be hidden / fixed in the background

**Review by simon-hng — APPROVED** [review]
Code changes look good

**Review by JohannesStoehr — APPROVED** [review]
Codechanges look good to me

**Review by tobias-lippert — COMMENTED** [review]
Tested in testing session on ts1. Worked as expected. Code looks good as well. Just saw branch has been merged anyway.

**Pull request merged**